nuxeo-features/nuxeo-automation/nuxeo-automation-io/src/main/java/org/nuxeo/ecm/automation/jaxrs/io/JsonRecordSetWriter.java

```diff
@@ -84,43 +84,44 @@ public void writeTo(RecordSet records, Class<?> type, Type genericType, Annotati
 
     protected void writeRecords(OutputStream out, RecordSet records) throws IOException {
 
-        JsonGenerator jg = factory.createJsonGenerator(out, JsonEncoding.UTF8);
-
-        jg.writeStartObject();
-        jg.writeStringField("entity-type", "recordSet");
-
-        if (records instanceof PaginableRecordSet) {
-            PaginableRecordSet pRecord = (PaginableRecordSet) records;
-            jg.writeBooleanField("isPaginable", true);
-            jg.writeNumberField("resultsCount", pRecord.getResultsCount());
-            jg.writeNumberField("pageSize", pRecord.getPageSize());
-            jg.writeNumberField("maxPageSize", pRecord.getMaxPageSize());
-            jg.writeNumberField("currentPageSize", pRecord.getCurrentPageSize());
-            jg.writeNumberField("currentPageIndex", pRecord.getCurrentPageIndex());
-            jg.writeNumberField("numberOfPages", pRecord.getNumberOfPages());
-            jg.writeBooleanField("isPreviousPageAvailable", pRecord.isPreviousPageAvailable());
-            jg.writeBooleanField("isNextPageAvailable", pRecord.isNextPageAvailable());
-            jg.writeBooleanField("isLastPageAvailable", pRecord.isLastPageAvailable());
-            jg.writeBooleanField("isSortable", pRecord.isSortable());
-            jg.writeBooleanField("hasError", pRecord.hasError());
-            jg.writeStringField("errorMessage", pRecord.getErrorMessage());
-        }
+        try (JsonGenerator jg = factory.createGenerator(out, JsonEncoding.UTF8)) {
```

Comment:
No longer calling the js.flush() method.  The javadocs says `that generator will NOT close the output stream when JsonGenerator#close is called (unless auto-closing feature is enabled`.  Is that the case?

Reply:
Yes AUTO_CLOSE_TARGET is always true by default. It's the normal behavior. Only in specialized cases would you want to not close the target when the generator is closed.
Explicit flush has been removed because it's not needed, as it's implicit in the close (I checked the implementation).